components/client/src/test/unit/java/com/hotels/styx/client/StyxHttpClientTest.java

```diff
@@ -54,208 +54,281 @@
+                .send(secureRequest)
+                .get();
 
-        await(client.sendRequest(request));
+        assertThat(response.status(), is(OK));
     }
 
     @Test
-    public void sendsToDefaultHttpPort() {
-        Connection.Factory connectionFactory = mockConnectionFactory(mockConnection(response(OK).build()));
-
+    public void overridesTlsSettingsWithSecureBoolean() throws ExecutionException, InterruptedException {
         StyxHttpClient client = new StyxHttpClient.Builder()
-                .setConnectionFactory(connectionFactory)
                 .build();
 
-        client.sendRequest(get("/")
-                .header(HOST, "localhost")
-                .build());
-
-        ArgumentCaptor<Origin> originCaptor = ArgumentCaptor.forClass(Origin.class);
-        verify(connectionFactory).createConnection(originCaptor.capture(), any(ConnectionSettings.class));
+        FullHttpResponse response = client
+                .secure(true)
+                .send(secureRequest)
+                .get();
 
-        assertThat(originCaptor.getValue().port(), is(80));
+        assertThat(response.status(), is(OK));
     }
 
     @Test
-    public void sendsToDefaultHttpsPort() {
-        Connection.Factory connectionFactory = mockConnectionFactory(mockConnection(response(OK).build()));
-        TlsSettings tlsSettings = mock(TlsSettings.class);
-
+    public void overridesTlsSettingsWithSecureBooleanFalse() throws ExecutionException, InterruptedException {
         StyxHttpClient client = new StyxHttpClient.Builder()
-                .setConnectionFactory(connectionFactory)
-                .tlsSettings(tlsSettings)
+                .tlsSettings(new TlsSettings.Builder().build())
                 .build();
 
-        client.sendRequest(get("/")
-                .header(HOST, "localhost")
-                .build());
-
-        ArgumentCaptor<Origin> originCaptor = ArgumentCaptor.forClass(Origin.class);
-        verify(connectionFactory).createConnection(originCaptor.capture(), any(ConnectionSettings.class));
+        FullHttpResponse response = client
+                .secure(false)
+                .send(httpRequest)
+                .get();
 
-        assertThat(originCaptor.getValue().port(), is(443));
+        assertThat(response.status(), is(OK));
     }
 
-    private StyxHttpClient httpClient() {
-        return new StyxHttpClient.Builder()
+    @Test(expectedExceptions = NullPointerException.class)
+    public void requiresValidTlsSettins() {
+        new StyxHttpClient.Builder()
+                .tlsSettings(null)
                 .build();
     }
 
-    private StyxHttpClient httpsClient() {
-        return new StyxHttpClient.Builder()
-                .connectionSettings(new ConnectionSettings(1000))
-                .tlsSettings(new TlsSettings.Builder()
-                        .authenticate(false)
-                        .build())
-                .responseTimeoutMillis(6000)
-                .build();
+    // TODO: Mikko:
```

Comment:
`TODO` left in PR.

Reply:
Okay. It is there to remind you guys to offer your opinion.

I think this test case should be removed if it doesn't run 100% of the time. Now it seems to depend on your ISP if it runs or not.

Shall we just get rid of this test? @a-dlatorre @kvosper what do you think?

Reply:
Removed this test.